doc/user-guide.md

```diff
@@ -61,11 +61,32 @@ The Manager will automatically register the scheme for all custom resources defi
 
 The Manager can restrict the namespace that all controllers will watch for resources:
 ```Go
-mgr, err := manager.New(cfg, manager.Options{Namespace: namespace})
+// Create a new Cmd to provide shared dependencies and start components
+mgr, err := manager.New(cfg, manager.Options{
+   Namespace: namespace,
+   MapperProvider:     restmapper.NewDynamicRESTMapper,
+   MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
```

Comment:
IMO, we should not add the `MapperProvider` and `MetricsBindAddress` here (or below), since they aren't really relevant to the examples being presented, which are about setting up watches for namespace scoping.

Also for `MapperProvider`, in the relatively near future an [improved DynamicRESTMapper will be added to controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/pull/554) and become the default for the Manager. When that happens and we update our C-R dep, our docs will be outdated and suggesting a worse option if we don't remember to update them (which we probably wouldn't remember if we're being honest).

For `MetricsBindAddress`, I think the `Advanced Topics -> Metrics` section that links to our detailed metrics docs is sufficient for the User Guide, which IMO is meant to get users up and running with the bare minimum and then to link to other docs about the more advanced topics.